### DISCUSSION TITLE
socks节点服务器具备故障自动切换功能吗？

### DISCUSSION THREAD

**Discussion by shimiaoyerin** [Openwrt-Passwall/openwrt-passwall2#327] [General]
![QQ图片20230816132157](https://github.com/xiaorouji/openwrt-passwall2/assets/45694554/4445b0ba-8984-45af-a053-f42254cd8521)
如图中这个服务器，能否故障的时候能自动切换？
有人可能想问我为什么使用分流规则，其实是我需求用不着，因为如果我开了，所有流量都会经过passwall2，我路由器具备nss加速，我并不想所有流量跑CPU去，我只想特定的软件走代理就行了，所以才限制一个端口分流

**Comment by handsomeji** [comment]
最新代码已支持~~~

**Reply by shimiaoyerin** [reply]
有个问题很好奇噢，因为机场的节点经常失效，就需要经常的更新订阅链接，如果此时有一种情况，比如本来节点叫“美国01”，但是此时我有两个机场的节点都叫“美国01呢”？如果节点判断不是根据名字来判断是不是原来选择的节点，那么passwall2是怎么认出原来的“美国01”？会不会出现勾选了备用节点，但是节点更新订阅之后，他认不出备用节点的情况？

  我上诉描述是设想无人值守的弊端，因为订阅链接是可以保证长时间不变的，但是订阅节点的名称可能就很跳跃了，有可能随着时间的推移，节点的名称也会变化
  能不能有一个随机切换？就像clash的故障切换那样，在所有的可用节点中，随机选择一个，以此确保长时间的无人值守
